appium/webdriver/appium_connection.py

```diff
@@ -30,51 +30,39 @@
 class AppiumConnection(RemoteConnection):
     _proxy_url: Optional[str]
 
+    RemoteConnection.user_agent = f'{PREFIX_HEADER}{library_version()} ({RemoteConnection.user_agent})'
+
     def __init__(
         self,
         remote_server_addr: str,
-        keep_alive: bool = False,
+        keep_alive: Optional[bool] = False,
         ignore_proxy: Optional[bool] = False,
         init_args_for_pool_manager: Union[Dict[str, Any], None] = None,
+        client_config: Optional[ClientConfig] = None,
     ):
-        # Need to call before super().__init__ in order to pass arguments for the pool manager in the super.
-        self._init_args_for_pool_manager = init_args_for_pool_manager or {}
-
-        super().__init__(remote_server_addr, keep_alive=keep_alive, ignore_proxy=ignore_proxy)
-
-    def _get_connection_manager(self) -> Union[urllib3.PoolManager, urllib3.ProxyManager]:
-        # https://github.com/SeleniumHQ/selenium/blob/0e0194b0e52a34e7df4b841f1ed74506beea5c3e/py/selenium/webdriver/remote/remote_connection.py#L134
-        pool_manager_init_args = {'timeout': self.get_timeout()}
-
-        if self._ca_certs:
-            pool_manager_init_args['cert_reqs'] = 'CERT_REQUIRED'
-            pool_manager_init_args['ca_certs'] = self._ca_certs
-        else:
-            # This line is necessary to disable certificate verification
-            pool_manager_init_args['cert_reqs'] = 'CERT_NONE'
+        if client_config is None:
+            client_config = ClientConfig(remote_server_addr=remote_server_addr)
+        if keep_alive is not None:
+            client_config.keep_alive = keep_alive
```

Comment:
I am not quite sure if we want to mutate the passed client config. At least, this must be explicitly stated in the docs or in the comment